propagation.go

```diff
@@ -112,8 +112,27 @@ type TextMapReader interface {
 	ForeachKey(handler func(key, val string) error) error
 }
 
+// TextMapCarrier allows the use of regular map[string]string
+// as both TextMapWriter and TextMapReader.
```

Comment:
technically this is in violation of the multimap spec, though the multimap spec is about to change... what to do??